packages/grid/backend/grid/core/node.py

```diff
@@ -15,10 +18,74 @@
 from grid.db.session import get_db_engine
 from grid.db.session import get_db_session
 
+
+def thread_function(*args, **kwargs) -> None:  # type: ignore
+    # TODO: call this after the Network node is deployed instead of using a timer.
+    time.sleep(30)
+
+    # third party
+    from requests import get
+
+    ip = get("https://api.ipify.org").content.decode("utf8")
+    print(f"My public IP address is: {ip}")
+
+    try:
+
+        NETWORK_PUBLIC_HOST = "http://" + ip + ":80"
+        # syft absolute
+        import syft as sy
+
+        network_root = sy.login(
+            email="info@openmined.org",
+            password="changethis",
+            url="http://" + ip,
+            port=80,
+        )
+    except Exception:
+
+        try:
+
+            NETWORK_PUBLIC_HOST = "http://localhost:80"
+
+            network_root = sy.login(
+                email="info@openmined.org",
+                password="changethis",
+                url="http://localhost",
+                port=80,
+            )
+        except Exception:
+
+            NETWORK_PUBLIC_HOST = "http://docker-host:80"
+
+            network_root = sy.login(
+                email="info@openmined.org",
+                password="changethis",
+                url="http://docker-host",
+                port=80,
+            )
+
+    vpn_s = network_root.vpn_status()
+
+    # if the VPN is empty then it's ready for the network node to join it (first)
+    if len(vpn_s["host"]) == 0 and len(vpn_s["peers"]) == 0:
+        network_root.join_network(host_or_ip=NETWORK_PUBLIC_HOST)
+
+
 if settings.NODE_TYPE.lower() == "domain":
     node = Domain("Domain", db_engine=get_db_engine(), settings=settings)
 elif settings.NODE_TYPE.lower() == "network":
     node = Network("Network", db_engine=get_db_engine(), settings=settings)
+    format = "%(asctime)s: %(message)s"
+    logging.basicConfig(format=format, level=logging.INFO, datefmt="%H:%M:%S")
+
+    logging.info("Main    : before creating thread")
+    x = threading.Thread(target=thread_function)
```

Comment:
MAYBE: I think we could use green threads here as gevent monkey patching could be done at level, which does implicit context switches?